### PR TITLE
Login frontend 2

### PR DESCRIPTION
**Pull request by patofernandez**
## Summary
The RouteActivity class was created. There it is validated if there is a session started and according to the result starts the LoginActivity.
It is validated that the fields are not incomplete and the mail has a valid format
## Screenshot
image

## Trello
https://trello.com/c/n8XqHje6

### CODE REVIEW

File: app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java [hunk 1]
```diff
@@ -60,21 +66,44 @@ public void afterTextChanged(Editable s) {
         vLoginButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                Toast.makeText(getContext(), "Log In press", Toast.LENGTH_LONG).show();
-                //getPresenter().storeUsername(vUserNameInput.getText().toString());
+                if (TextUtils.isEmpty(vUserNameInput.getText())) {
```
Comment (DemianIac) [inline app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java]: Maybe we can move this to a new method. like IsValidUsernameAndPassword() that return a boolean to do something

File: app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java [hunk 2]
```diff
@@ -60,21 +66,44 @@ public void afterTextChanged(Editable s) {
...
             public void onClick(View v) {
+                vUserNameInput.requestFocus();
                 Toast.makeText(getContext(), "Sign In press", Toast.LENGTH_LONG).show();
             }
         });
 
         vTermsAndConditions.setMovementMethod(LinkMovementMethod.getInstance());
     }
 
+    private void guardarUsuario(String userMail) {
```
Comment (DemianIac) [inline app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java]: Change the name of the method and send the sharedPreference to the presenter.

**Review by DemianIac — CHANGES_REQUESTED** [review]

File: app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java (hunk 1 above)
Comment (patofernandez) [reply app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java]: Done!

File: app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java (hunk 2 above)
Comment (patofernandez) [reply app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java]: Done!

File: app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java [hunk 3]
```diff
@@ -60,21 +63,38 @@ public void afterTextChanged(Editable s) {
...
 
         vTermsAndConditions.setMovementMethod(LinkMovementMethod.getInstance());
     }
 
+    private boolean isValidUsernameAndPassword(String userName) {
+        return TextUtils.isEmpty(userName);
+    }
+
+    private boolean isValidEmail(String email) {
+        Pattern pattern = Patterns.EMAIL_ADDRESS;
```
Comment (DemianIac) [inline app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java:94]: You can write this in one line . return android.util.Patterns.EMAIL_ADDRESS.matcher(target).matches() . But it's okay for now

**Review by DemianIac — APPROVED** [review]

File: app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java (hunk 3 above)
Comment (patofernandez) [reply app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java:94]: Ok

**Pull request merged**